extensions/amp-story/1.0/amp-story.js

```diff
@@ -912,14 +912,16 @@ export class AmpStory extends AMP.BaseElement {
 
   /** @private */
   initializePages_() {
+    const storyPages = this.element.querySelectorAll('amp-story-page');
     const pageImplPromises = Array.prototype.map.call(
-        this.element.querySelectorAll('amp-story-page'),
+        storyPages,
         (pageEl, index) => {
           return pageEl.getImpl().then(pageImpl => {
             this.pages_[index] = pageImpl;
           });
         });
 
+    this.storeService_.dispatch(Action.SET_PAGES_COUNT, storyPages.length);
```

Comment:
Do we need this immediately?  Can we instead use `this.pages_.length` in a chained promise after the `Promise.all`?  This will allow us to keep the `pages_` array in sync with the pages count, in case we e.g. add or drop pages during initialization

Reply:
Done.